### ISSUE TITLE
[BUG] GR `guidefontvalign`, `guidefonthalign` do not align the labels

### ISSUE THREAD

**Issue by mmikhasenko** [JuliaPlots/Plots.jl#3376]
```julia
plot(rand(10), xlab="x (Unt)", ylab="y (Unt)",
    guidefontvalign=:top, guidefonthalign=:right)
```
![image](https://user-images.githubusercontent.com/22725744/112459184-da5dfe00-8d5d-11eb-85af-2e9e7d0c6efa.png)

The issues seems to be in `gr_label_axis` function that 
https://github.com/JuliaPlots/Plots.jl/blob/c0824bdc628345b4859ebcafc437cbfd63e63ccb/src/backends/gr.jl#L1485

Versions:
```julia
[28b8d3ca] GR v0.55.0 `https://github.com/jheinen/GR.jl.git#master`
[d2c73de3] GR_jll v0.53.0+0
>>>>>
Julia Version 1.5.0
Commit 96786e22cc (2020-08-01 23:44 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, skylake)
Environment:
  JULIA_EDITOR = "Microsoft VS Code\Code.exe"
  JULIA_NUM_THREADS = 1
```

**Comment by mmikhasenko** [comment]
A dirty fix
```julia
gr_text(gr_view_xright(viewport_plotarea), viewport_plotarea[3] - h, xaxis[:guide])
gr_text(viewport_plotarea[1] - w, gr_view_yright(viewport_plotarea), yaxis[:guide])
```
with
```julia
gr_view_xright(viewport_plotarea) = viewport_plotarea[1] + 0.9 * (viewport_plotarea[2] - viewport_plotarea[1])
gr_view_yright(viewport_plotarea) = viewport_plotarea[3] + 0.9 * (viewport_plotarea[4] - viewport_plotarea[3])
```

**Comment by mmikhasenko** [comment]
A bit more accurate
```julia
gr_view_xposition(viewport_plotarea, position) = viewport_plotarea[1] + position * (viewport_plotarea[2] - viewport_plotarea[1])
gr_view_yposition(viewport_plotarea, position) = viewport_plotarea[3] + position * (viewport_plotarea[4] - viewport_plotarea[3])

function position(symb)
    if symb == :top || symb == :right
        return 0.95
    elseif symb == :left || symb == :bottom
        return 0.05
    end
    return 0.5
end

function alignment(symb)
    if symb == :top || symb == :right
        return GR.TEXT_HALIGN_RIGHT
    elseif symb == :left || symb == :bottom
        return GR.TEXT_HALIGN_LEFT
    end
    return GR.TEXT_HALIGN_CENTER
end
```

**Comment by mmikhasenko** [comment]
fixed for `gr()` by #3377 #3390
